### PR TITLE
Bug(401603): Remove editor button when form is live

### PR DESCRIPTION
**Pull request by mokhld**
This PR removes the `Editor` button when a form is live. It also increases test coverage by adding a test suite for `library.js`.

### CODE REVIEW

File: designer/server/src/models/forms/form-lifecycle.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ export function confirmationPageViewModel(form, errorList) {
   const pageTitle = 'Are you sure you want to make the draft live?'
 
   const formPath = formOverviewPath(form.slug)
-  const navigation = getFormSpecificNavigation(formPath)
+  const navigation = getFormSpecificNavigation(formPath, form, '')
```
Comment (colinrotherham) [inline designer/server/src/models/forms/form-lifecycle.js:14]: Do we need this if `''` is the default?

```suggestion
  const navigation = getFormSpecificNavigation(formPath, form)
```

Comment (mokhld) [reply designer/server/src/models/forms/form-lifecycle.js:14]: Good spot 👍

Comment (mokhld) [reply designer/server/src/models/forms/form-lifecycle.js]: Pushed.

File: designer/server/src/models/forms/library.test.js [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+      createdAt: now,
+      createdBy: liveAuthor,
+      updatedAt: now,
+      updatedBy: liveAuthor
+    }
+  }
+
+  describe('getFormSpecificNavigation', () => {
+    describe('with draft existing', () => {
+      test('includes Editor link', () => {
```
[the remark this answers is not in the document]
Comment (mokhld) [reply designer/server/src/models/forms/library.test.js]: Pushed.

[the remark this answers is not in the document]
Comment (colinrotherham) [reply designer/server/src/models/forms/library.test.js]: Yeah exactly the same

Think we just have more `it('should...')` than `test('if...')` currently

[the remark this answers is not in the document]
Comment (mokhld) [reply designer/server/src/models/forms/library.test.js]: Yeah that makes sense tbf- definitely a lot more readable.

**Review by colinrotherham — APPROVED** [review]

**Pull request merged**